src/BinSkim.Rules/PERules/BA6001.EnableSourceLink.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+
+using Microsoft.CodeAnalysis.BinaryParsers;
+using Microsoft.CodeAnalysis.BinaryParsers.ProgramDatabase;
+using Microsoft.CodeAnalysis.IL.Sdk;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.IL.Rules
+{
+    [Export(typeof(Skimmer<BinaryAnalyzerContext>)), Export(typeof(ReportingDescriptor)), Export(typeof(IOptionsProvider))]
```

Comment:
>IOptionsProvider [](http://example.com/codeflow?start=104&length=16)

Strictly speaking you don't need IOptionsProvider support because we have no knobs (yet). :) Have you thought about whether this check might benefit from some configurability? Nothing occurs to me yet, thought I'd ask. If we get to merge and your are still returning an empty array, let's zap your IOptionsProvider support. The reason is that binskim supports an export-config option that produces a default XML config file (to be tweaked by users to configure config-driven analysis). If you implement this interface, that command will find you and potentially record some data in the doc where none is expected.

btw - I may be wrong about that. As I recall, the export-config *always* has a record for every rule, so that you can configure turning it off or changing severity. So it may handle an empty array of options just fine. If you have a moment to check it out, let me know what you find. Isn't important.

Reply:
Removed the IOptionsProvider